cpp/src/lance/arrow/scanner.cc

```diff
@@ -67,37 +65,18 @@ ::arrow::Result<std::shared_ptr<::arrow::dataset::Scanner>> ScannerBuilder::Fini
     auto schema = lance::format::Schema(scanner->options()->dataset_schema);
     auto columns = columns_.value();
 
-    if (::arrow::compute::ExpressionHasFieldRefs(scanner->options()->filter)) {
-      for (auto& filtered_column :
-           ::arrow::compute::FieldsInExpression(scanner->options()->filter)) {
-        auto filtered_col_name = ColumnNameFromFieldRef(filtered_column);
-        if (std::find(std::begin(columns), std::end(columns), filtered_col_name) == columns.end()) {
-          columns.emplace_back(filtered_col_name);
-        }
-      }
-    }
     ARROW_ASSIGN_OR_RAISE(auto projected_schema, schema.Project(columns));
-    scanner->options()->dataset_schema = projected_schema->ToArrow();
     ARROW_ASSIGN_OR_RAISE(scanner->options()->filter,
                           scanner->options()->filter.Bind(*scanner->options()->dataset_schema));
+    scanner->options()->projected_schema = projected_schema->ToArrow();
```

Comment:
Does it matter whether this is done before or after the filter bind?

Reply:
the filter bined on dataset schema, which changes only the filter internal.

This assigntment is over project schema, which is separated from dataset schema. I think it does not matter to the filter binding order.